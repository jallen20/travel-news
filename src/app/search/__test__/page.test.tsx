import {describe, expect, it} from "@jest/globals";
import SearchPage from "../page";
import {render} from "@testing-library/react";

jest.mock("next/navigation", () => ({
    useRouter() {
        return {
            route: "/",
            pathname: "",
            query: "",
            asPath: "",
        };
    },
}));

describe("Search Page Tests", () => {
    it("Should display no results", () => {
        const {container} = render(
            <SearchPage />,
            
        );
        expect(container).toMatchSnapshot();
    });
})
