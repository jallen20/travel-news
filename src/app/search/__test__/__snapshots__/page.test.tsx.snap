// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Page Tests Should display no results 1`] = `
<div>
  <div
    class="search-page"
  >
    <div
      class="search-bar-wrapper"
    >
      <form>
        <input
          class="search-bar"
          type="text"
          value=""
        />
        <button
          type="submit"
        >
          <svg
            fill="currentColor"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11.742 10.344a6.5 6.5 0 1 0-1.397 1.398h-.001c.03.04.062.078.098.115l3.85 3.85a1 1 0 0 0 1.415-1.414l-3.85-3.85a1.007 1.007 0 0 0-.115-.1zM12 6.5a5.5 5.5 0 1 1-11 0 5.5 5.5 0 0 1 11 0z"
            />
          </svg>
        </button>
      </form>
    </div>
    <div
      class="no-results"
    >
      <div
        class="no-results-content"
      >
        <svg
          fill="currentColor"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M11.742 10.344a6.5 6.5 0 1 0-1.397 1.398h-.001c.03.04.062.078.098.115l3.85 3.85a1 1 0 0 0 1.415-1.414l-3.85-3.85a1.007 1.007 0 0 0-.115-.1zM12 6.5a5.5 5.5 0 1 1-11 0 5.5 5.5 0 0 1 11 0z"
          />
        </svg>
        <span>
          No Results Found.
        </span>
      </div>
    </div>
  </div>
</div>
`;
